quickstart/aspnet/ContosoTeamStats/RedisConnection.cs

```diff
@@ -0,0 +1,216 @@
+        private ConnectionMultiplexer _connection;
+        private IDatabase _database;
+        private bool _didInitialize = false;
+
+        public async Task InitializeAsync()
+        {
+            if (!_didInitialize)
+            {
+                _didInitialize = true;
+                _connection = await CreateConnectionAsync();
+                _database = _connection.GetDatabase();
+            }
+        }
+
+        // In real applications, consider using a framework such as
+        // Polly to make it easier to customize the retry approach.
+        // For more info, please see: https://github.com/App-vNext/Polly
+        public async Task<T> BasicRetryAsync<T>(Func<IDatabase, Task<T>> func)
+        {
+            int reconnectRetry = 0;
+
+            while (true)
+            {
+                try
+                {
+                    return await func(_database);
+                }
+                catch (Exception ex) when (ex is RedisConnectionException || ex is SocketException)
+                {
+                    reconnectRetry++;
+                    if (reconnectRetry > _retryMaxAttempts)
+                        throw;
+                    try
+                    {
+                        await ForceReconnectAsync();
+                    }
+                    catch (ObjectDisposedException) { }
+                }
+            }
+        }
+
+        // This method may return null if it fails to acquire the semaphore in time.
+        // Use the return value to update the "connection" field
+        private async Task<ConnectionMultiplexer> CreateConnectionAsync()
+        {
+            if (_connection != null)
+            {
+                // If we already have a good connection, let's re-use it
+                return _connection;
+            }
+
+            try
+            {
+                await _initSemaphore.WaitAsync(_restartConnectionTimeout);
+            }
+            catch
+            {
+                // We failed to enter the semaphore in the given amount of time. Connection will either be null, or have a value that was created by another thread.
+                return _connection;
+            }
+
+            // We entered the semaphore successfully.
+            try
+            {
+                if (_connection != null)
+                {
+                    // Another thread must have finished creating a new connection while we were waiting to enter the semaphore. Let's use it
+                    return _connection;
+                }
+
+                // Otherwise, we really need to create a new connection.
+                string cacheConnection = @"michelle-standard-jan6.redis.cache.windows.net:6380,password=9LgWRLM8fU08GSChWNuCh27H8F9W8YiSbAzCaIIxS6E=,ssl=True,abortConnect=False";// ConfigurationManager.AppSettings["CacheConnection"].ToString();
+                return await ConnectionMultiplexer.ConnectAsync(cacheConnection);
```

Comment:
pretty sure you want to back this out :)
(and rotate your cache keys, now that this has been pushed to github)